catkin_ws/src/barracuda_thruster_output_controller/scripts/rpi_thruster_controller.py

```diff
@@ -45,13 +49,36 @@ def on_recv_thruster_kgf(msg, thruster_id):
     
     send_duty_cycle_val_to_thruster(duty_cycle_val, thruster_id)
     
+# software kill switch service to stop thrusters
+def set_thruster_zero():
+    service_name = '/set_thrust_zero'
+    rospy.wait_for_service(service_name)
+    try:
+        set_thruster_zero_srv = rospy.ServiceProxy(service_name, SetThrustZero)
+        resp = set_thruster_zero_srv()  # no arguments since request is empty
+        if resp.success:
+            print(f"Service call succeeded: {resp.message}")
+        else:
+            print(f"Service call failed: {resp.message}")
+    except rospy.ServiceException as e:
+        print(f"Service call failed: {e}")
+
+def check_disable_pin(event):
+    if GPIO.input(DISABLE_PIN) == GPIO.LOW:
```

Comment:
you need to call the service with an argument (see service definition)

and for better performance use an edge detector to call the service. handle edge cases (i.e. set the value at the beginning when there is no edge)